backend/src/main/java/com/myshop/products/ProductController.java

```diff
@@ -21,46 +22,80 @@ public class ProductController {
     private final ProductService productService;
 
     @GetMapping
-    @ResponseBody
-    public ResponseEntity<ResponseData> getAllProduct(@RequestParam(required = false) String name) {
-        //Get product by searching name
-        if (name != null) {
-            //change name into nameId format
+    public ResponseEntity<ResponseData> getAllProduct() {
+        try {
+            List<Product> products = productService.getAllProduct();
+            return new ResponseEntity<>(new SuccessfulResponse<>(products.size(), products), HttpStatus.OK);
+
+        } catch (Exception e) {
+            return new ResponseEntity<>(new FailureResponse("Internal Server Error"), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GetMapping(params = {"name"})
+    public ResponseEntity<ResponseData> getAllProductOfName(@RequestParam String name) {
+        try {
+            List<Product> products;
             String formattedSearchName = SLUGIFY.toSlug(name);
-            List<Product> filteredProducts = productService.findProductByName(formattedSearchName);
-            // In case find result and not empty List
-            if (filteredProducts != null && filteredProducts.size() > 0)
-                return new ResponseEntity<>(new SuccessfulResponse<>(filteredProducts.size(), filteredProducts), HttpStatus.OK);
+            products = productService.findProductByName(formattedSearchName);
+
+            if (products == null || products.size() == 0)
+                return new ResponseEntity<>(new FailureResponse("Not found"), HttpStatus.NOT_FOUND);
 
-            return new ResponseEntity<>(new FailureResponse("Not found"), HttpStatus.NOT_FOUND);
+            return new ResponseEntity<>(new SuccessfulResponse<>(products.size(), products), HttpStatus.OK);
+
+        } catch (Exception e) {
+            return new ResponseEntity<>(new FailureResponse("Internal Server Error"), HttpStatus.INTERNAL_SERVER_ERROR);
         }
+    }
 
-        // get all product
+    @GetMapping(params = {"page","size"})
+    public ResponseEntity<ResponseData> getAllProductWithPaginate(@RequestParam String page, @RequestParam String size) {
```

Comment:
i don't think this is the correct Type for page and size parameter.
 what will happen if i use: **Page** = '**a12**' and **size** = '**2b3**'

Reply:
In productService, I tried to parse these parameters to int. So with the case **Page = 'a12' and size = '2b3'**,  DEFAULT_VALUE will be applied to them.